packages/web/src/components/gcds-site-menu/gcds-site-menu.tsx

```diff
@@ -240,145 +92,79 @@ export class GcdsSiteMenu {
+        await unpackMobileMenu(menu);
+        await menu.updateMenuItemQueue(menu);
       } else {
-        if (mobileLayout == "drawer") {
-          document.querySelector("body").style.paddingBottom = "3rem";
-        }
-
-        if (desktopLayout == "sidebar" && !hostElement.shadowRoot.querySelector("[data-sidebar-backdrop]").hasAttribute("hidden")) {
-          hostElement.shadowRoot.querySelector("[data-sidebar-backdrop]").setAttribute("hidden", "");
-          document.querySelector("body").style.removeProperty("overflow");
-        }
-
-        hostElement.shadowRoot.querySelector("[data-h2-menu-container]").setAttribute("data-mobile", "");
-        h2MenuCloseOpenSubmenusHandler(hostElement);
+        menu.updateMenuSize("mobile");
+        await configureMobileMenu(menu);
+        await menu.updateMenuItemQueue(menu);
       }
     });
-
-    h2MenuAddUpDownArrowsToMainMenuItems(this.el);
-    h2MenuTabOrder(this.el);
-    h2MenuAddRightArrowToMainMenuItems(this.el);
-    h2MenuEnableSubmenuTriggers(this.el);
-    h2MenuAddMobileMenuTrigger(this.el);
-    h2MenuAddPageAnchor(this.el);
-
-    if (this.desktopLayout == "sidebar") {
-      menuEnableBackButtonTriggers(this.el);
-    } else {
-      this.el.addEventListener('focusout', () => {
-        if (!this.el.shadowRoot.querySelector('[data-mobile]')) {
-          h2MenuCloseOpenSubmenusHandler(this.el);
-        }
-      });
-    }
   }
 
   render() {
-    const sticky = this.position == 'sticky' ? true : false;
-    const mobileMenutask = this.mobileLayout == 'drawer' ?
-      <gcds-button
-        aria-expanded="false"
-        aria-label={I18N[this.lang].mobileTriggerLabel}
-        aria-haspopup="true"
-        class="mobile-menu-trigger"
-        type="button" role="button" button-role="primary">
-        <span data-h2-mobile-menu-trigger-open-label>Menu</span><span data-h2-mobile-menu-trigger-close-label>{I18N[this.lang].mobileTriggerClose}</span>
-      </gcds-button>
-      : '';
-
+    const { label, alignment } = this;
     return (
       <Host
-        data-h2-menu-wrapper
-        desktop-layout={this.desktopLayout}
-        mobile-layout={this.mobileLayout}
-        menu-sticky={sticky}
-        lang={this.lang}
+        role="navigation"
+        aria-label={label}
       >
-        {mobileMenutask}
-
-        <nav
-          aria-label={I18N[this.lang].navLabel}
-          data-h2-menu
-        >
-          <div data-h2-menu-container>
-            <div data-optional-left>
-              <slot name="left" />
-            </div>
-            <slot />
-            <div data-optional-right>
-              <slot name="right" />
-            </div>
-          </div>
-        </nav>
-
-        {this.desktopLayout == "sidebar" ?
-          <div data-sidebar-backdrop hidden onClick={() => {h2MenuCloseOpenSubmenusHandler(this.el)}}></div>
-        : null }
-
-        <slot name="main" />
+        <div class="gcds-site-menu__container">
```

Comment:
Can we use a `nav` tag here instead of the div to be more semantic?

Reply:
Same reasoning as above as to why it's not a `nav` element but open to change it

Reply:
Same answer as above :)